### ISSUE TITLE
Conditions / example of DepSepConvolution fusion

### ISSUE THREAD

**Issue by vadimkantorov** [NVIDIA/TensorRT#3237]
https://docs.nvidia.com/deeplearning/tensorrt/developer-guide/index.html#fusion-types says:

**Depthwise Separable Convolution**
`A depthwise convolution with activation followed by a convolution with activation `**may sometimes**` be fused into a single optimized DepSepConvolution layer. The precision of both convolutions must be INT8 and the device's computes capability must be 7.2 or later.`

Are there any other conditions? What types of activations are admissible?

Is there example of fusable graphs? (this is important especially given that convs must already be int8)

There is almost no example or mentions of DepSepConvolution/TRT in Google Search.

Wonder about constraints of Q-DQ and qparams.

Thank you :)

**Comment by zerollzeng** [comment]
@nvpohanh ^ ^

**Comment by nvpohanh** [comment]
You need: a pair of Q/DQ before depthwise Conv, a pair of Q/DQ before the 1x1 Conv, and a pair of Q/DQ before the next Conv (after 1x1 Conv's activation).

Here is an illustration:
![2023-08-23 12_11_45-C__Users_phuan_AppData_Local_Temp_MicrosoftEdgeDownloads_9490e67b-d4dc-45b7-9364](https://github.com/NVIDIA/TensorRT/assets/53919306/93eb8990-7a0e-4aaf-8718-d19f44cd0247)

**Comment by vadimkantorov** [comment]
Thank you! We will try this pattern!

It would be awesome to have this fusion example as an `.onnx` file and maybe a `.svg` output from trex.

**Comment by ttyio** [comment]
closing since no activity for more than 3 weeks, pls reopen if you still have question, thanks all!

**Comment by nvpohanh** [comment]
reopen for now. thanks

**Comment by nvpohanh** [comment]
@aboubezari Could you provide the ONNX file so that we can repro and debug this issue? Thanks

**Comment by aboubezari** [comment]
Yes, I've attached the ONNX file as a zip file with just the onnx model in it.
Let me know if you would like me to export different shapes or activations on the Convs. I have already tried using Relu activations instead of BatchNorm with no luck.
[aboubezari_debug.zip](https://github.com/NVIDIA/TensorRT/files/13944368/aboubezari_debug.zip)

**Comment by nvpohanh** [comment]
Filed internal tracker 4454538. Will let you know if we have any findings.

**Comment by aboubezari** [comment]
Awesome, thanks.

**Comment by nzmora-nvidia** [comment]
@aboubezari unrelated to the problem you've reported, I recommend placing the first BatchNorm after the first convolution (as it appears in the [diagram above](https://github.com/NVIDIA/TensorRT/issues/3237#issuecomment-1892856532)).
The ONNX file in [aboubezari_debug.zip](https://github.com/NVIDIA/TensorRT/files/13944368/aboubezari_debug.zip) looks like so:
![image](https://github.com/NVIDIA/TensorRT/assets/96238833/4bbbdccb-0d0d-4329-a64b-889ef221e483)

**Comment by aboubezari** [comment]
@nzmora-nvidia I realized that I exported the model after tweaking it a bit to figure out the issue, my bad. 
Let me know if you need me to export you a new model.

**Comment by nzmora-nvidia** [comment]
@aboubezari Thank you, we can recreate the error and do not need the new model.

**Comment by vadimkantorov** [comment]
> The ONNX file in [aboubezari_debug.zip](https://github.com/NVIDIA/TensorRT/files/13944368/aboubezari_debug.zip) looks like so:

I guess it would be awesome to have such example ONNX files (or even complete PyTorch + torch-tensorrt) examples in the docs of TRT, especially when fusion is discussed (and given that fusion patterns are often fragile, especially together with quantization)!

**Comment by nzmora-nvidia** [comment]
@vadimkantorov That's a fair request. I'll provide some pytorch examples in the next TREx release.

**Comment by vadimkantorov** [comment]
@nvpohanh Please add somewhere in the docs an example `*.onnx` file or PyTorch example of properly getting DepSepConv to be used in TRT :) This is a very important module for speed-ups, it's important for users to know how export recognizable patterns for it...

E.g. a complete example of export of MobileNetV3 (making use of DepSep) https://pytorch.org/vision/stable/models/generated/torchvision.models.quantization.mobilenet_v3_large.html#mobilenet-v3-large would be great

**Comment by aboubezari** [comment]
Thank you @nvpohanh! Look forward to trying it out.

**Comment by ttyio** [comment]
I will close this since this is solved, thanks all!
